metaflow/plugins/kfp/kfp_decorator.py

```diff
@@ -0,0 +1,89 @@
+import json
+import os
+import tarfile
+import tempfile
+from urllib.parse import urlparse
+
+from metaflow import util, current
+from metaflow.datastore import MetaflowDataStore
+from metaflow.datastore.util.s3util import get_s3_client
+from metaflow.decorators import StepDecorator
+from metaflow.metadata import MetaDatum
+from metaflow.metaflow_config import DATASTORE_LOCAL_DIR
+
+
+class KfpInternalDecorator(StepDecorator):
+    name = "kfp_internal"
+
+    def task_pre_step(
+        self,
+        step_name,
+        ds,
+        metadata,
+        run_id,
+        task_id,
+        flow,
+        graph,
+        retry_count,
+        max_retries,
+    ):
+        """
+        Analogous to step_functions_decorator.py
+        """
+        # TODO: any other KFP environment variables to get and register to Metadata service?
+        meta = {"kfp-execution": os.environ["METAFLOW_RUN_ID"]}
+        entries = [MetaDatum(field=k, value=v, type=k) for k, v in meta.items()]
+        # Register book-keeping metadata for debugging.
+        metadata.register_metadata(run_id, step_name, task_id, entries)
```

Comment:
Is this metadata going to the MF metadata store?

Reply:
Yes.  In our current KFP installation it is going to the local MF store, and in future it would goto the Metadata service from which it can store in any backend we wish.

Reply:
Which backends are we targeting? Are we still planning on spinning up a remote MF metadata store separate from the wider Metadata effort I think Kenneth is leading (or is that his recommendation)? Sorry if you don't know, I'm a bit out of the loop on this workstream!